apps/ledger-live-desktop/src/renderer/families/casper/TransferIdField.jsx

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import React, { useCallback } from "react";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import Input from "~/renderer/components/Input";
+import invariant from "invariant";
+import type { Account } from "@ledgerhq/types-live";
+import type { Transaction, TransactionStatus } from "@ledgerhq/live-common/families/casper/types";
+import { useTranslation } from "react-i18next";
+
+const TranferIdField = ({
+  onChange,
+  account,
+  transaction,
+  status,
+}: {
+  onChange: string => void,
+  account: Account,
+  transaction: Transaction,
+  status: TransactionStatus,
+}) => {
+  invariant(transaction.family === "casper", "TransferIdField: casper family expected");
+
+  const { t } = useTranslation();
+
+  const bridge = getAccountBridge(account);
+
+  const onTransferIdFieldChange = useCallback(
+    (value: string) => {
+      value = value.replace(/\D/g, "");
+      if (value !== "") onChange(bridge.updateTransaction(transaction, { transferId: value }));
+      else onChange(bridge.updateTransaction(transaction, { transferId: undefined }));
+    },
+    [onChange, transaction, bridge],
+  );
+
+  // We use transaction as an error here.
+  // on the ledger-live mobile
```

Comment:
Nice try :) but you don't set any `errors.transaction` or `warnings.transaction` in the `getTransactionStatus` function.

Reply:
haha, eye of an eagle. Fixed, poor me 😅